core/src/main/scala/kafka/controller/ControllerChannelManager.scala

```diff
@@ -138,9 +138,11 @@ class ControllerChannelManager(controllerContext: ControllerContext, config: Kaf
 
   private def removeExistingBroker(brokerState: ControllerBrokerStateInfo) {
     try {
+      // shutdown the RequestSendThread before closing the NetworkClient to
+      // avoid the concurrent use of the non-threadsafe classes as described in KAFKA-4959
+      brokerState.requestSendThread.shutdown()
```

Comment:
My bad, you are right. It was clear to me that there was no concurrent access, but I hadn't realised that we had a happens before edge via the `CountDownLatch`. Can we mention this in the comment?

Also, it seems like messageQueue is accessed by `RequestSendThread` and the thread queuing the requests without a shared lock. And the implementation is not thread-safe. Is this right or am I missing something?